village/src/App.js

```diff
@@ -14,11 +19,55 @@ class App extends Component {
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+  componentDidMount() {
+    axios
+      .get('http://localhost:3333/smurfs')
+      .then((response) => {
+        this.setState({ smurfs: response.data });
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  addNewSmurf = (smurf) => {
+    axios
+      .post('http://localhost:3333/smurfs', smurf)
+      .then((response) => this.setState({ smurfs: response.data }));
+  };
+
+  deleteSmurf = (smurfId) => {
+    return axios
+      .delete(`http://localhost:3333/smurfs/${smurfId}`)
+      .then((response) => this.setState({ smurfs: response.data }));
+  };
+
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <Nav />
+        <Route exact path="/" component={Header} />
+        <Route
+          exact
+          path="/smurfs"
+          render={(props) => (
+            <Smurfs
+              {...props}
+              smurfs={this.state.smurfs}
+              deleteSmurf={this.deleteSmurf}
+            />
+          )}
+        />
+
+        <Route
+          path="/addsmurf"
+          render={(props) => <SmurfForm addNewSmurf={this.addNewSmurf} />}
+        />
+
+        <Route
+          path="/smurfs/:smurfId"
+          render={(props) => <Smurf {...props} smurfs={this.state.smurfs} />}
+        />
```

Comment:
The organization of this file is great! The only suggestion I can give is to look at your props and data that are being passed around, and decide if you would change anything there.